chain/test/Test/Pos/Chain/Txp/Arbitrary.hs

```diff
@@ -67,6 +78,42 @@ instance Arbitrary TxSigData where
     arbitrary = genericArbitrary
     shrink = genericShrink
 
+-- This instance avoids the failure of tests that
+-- use checkTx but aren't explicitly checking if
+-- if checkTx is behaving as it should.
+instance Arbitrary TxValidationRules where
+    arbitrary = do
+        cutoffEpoch <- arbitrary
+        pure $ TxValidationRules (EpochOrSlot . Left $ EpochIndex cutoffEpoch)
+                                 (EpochOrSlot . Left $ EpochIndex (cutoffEpoch - 1))
+                                 1
```

Comment:
Are we unable to use `arbitrary` here?

Reply:
We can't cause other tests that use `TxValidationRules`  generate values that will fail in `checkTx`. I wrote tests that use hardcoded `TxValidationRules` values to be sure `checkTx` is working appropriately. So this instance is just to avoid failures in other tests that aren't testing this property.

Reply:
👍 adding a note for posterity:

`TxValidationRules` includes the epoch in which AddrAttr size limitation is enforced, and the current epoch. The datatype is constructed fresh for each epoch (possibly each slot but at least each epoch) and thus it informs the validation code of whether the cutoff has been passed.

The above `arbitrary` instance is generating a random `EpochIndex` as the cutoff, and telling us that we're in the epoch preceding the cutoff. So tx validation code shouldn't enforce the size restriction because it would go into effect on the next epoch.